src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -128,15 +89,56 @@ uct_cuda_ipc_mem_add_reg(void *addr, uct_cuda_ipc_memh_t *memh,
 
         key->ph.handle_type = UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM;
         ucs_trace("packed vmm fabric handle for %p", addr);
+        goto common_path;
+    } else {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(&mempool,
+                    CU_POINTER_ATTRIBUTE_MEMPOOL_HANDLE, (CUdeviceptr)addr));
+        if (status == UCS_OK) {
+            status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(
+                        &allowed_handle_types,
+                        CU_POINTER_ATTRIBUTE_ALLOWED_HANDLE_TYPES,
+                        (CUdeviceptr)addr));
+            if ((status != UCS_OK) ||
+                (!(allowed_handle_types | CU_MEM_HANDLE_TYPE_FABRIC))) {
+                status = UCS_ERR_NO_RESOURCE;
+                goto err;
+            }
+
+            status =
+                UCT_CUDADRV_FUNC_LOG_DEBUG(cuMemPoolExportToShareableHandle(
+                            (void *)&key->ph.handle.fabric_handle, mempool,
+                            CU_MEM_HANDLE_TYPE_FABRIC, 0));
+            if (UCS_OK != status) {
+                /* VMM, mempool queries failed. Try legacy IPC handle */
```

Comment:
seems we should get here only if CU_MEM_HANDLE_TYPE_FABRIC is included in allowed_handle_types, so the fallback to legacy should happen when checking allowed_handle_types and not here. Or no need to check allowed_handle_types at all.